app/src/main/scala/org/alephium/explorer/persistence/dao/BlockDao.scala

```diff
@@ -95,7 +95,40 @@ object BlockDao {
       .newBuilder()
       .maximumSize(groupConfig.chainNum.toLong)
       .expireAfterWrite(5, java.util.concurrent.TimeUnit.SECONDS)
-      .buildAsync(asyncLoader)
+      .buildAsync(latestBlockAsyncLoader)
+
+    private val blockTimeAsyncLoader: AsyncCacheLoader[ChainIndex, Duration] = {
+      case (key, _) =>
+        val chainFrom = GroupIndex.fromProtocol(key.from)
+        val chainTo   = GroupIndex.fromProtocol(key.to)
+        (for {
+          latestBlock <- cachedLatestBlocks.get(key).asScala
+          after = latestBlock.timestamp.minusUnsafe(Duration.ofHoursUnsafe(2))
+          blockTimes <- run(getBlockTimes(chainFrom, chainTo, after))
+        } yield {
+          computeAverageBlockTime(blockTimes)
+        }).asJava.toCompletableFuture
+    }
+
+    @SuppressWarnings(Array("org.wartremover.warts.TraversableOps"))
+    private def computeAverageBlockTime(blockTimes: Seq[TimeStamp]): Duration = {
+      if (blockTimes.size > 1) {
+        val (_, diffs) =
+          blockTimes.drop(1).foldLeft((blockTimes.head, Seq.empty: Seq[Duration])) {
+            case ((prev, acc), ts) =>
+              (ts, acc :+ (ts.deltaUnsafe(prev)))
+          }
+        diffs.fold(Duration.zero)(_ + _).divUnsafe(diffs.size.toLong)
+      } else {
+        Duration.zero
+      }
+    }
+
+    private val cachedBlockTimes: AsyncLoadingCache[ChainIndex, Duration] = Caffeine
+      .newBuilder()
+      .maximumSize(groupConfig.chainNum.toLong)
+      .expireAfterWrite(5, java.util.concurrent.TimeUnit.SECONDS)
+      .buildAsync(blockTimeAsyncLoader)
```

Comment:
Since `cachedBlockTimes` and `latestBlockAsyncLoader` have the same configuration and keys do you think we could use a single cache or do they have to be seperate?

I'm totally ok with this. I couldn't tell single which way is better without profiling :|